backend/src/main/kotlin/com/infowings/catalog/external/ObjectApi.kt

```diff
@@ -24,6 +25,21 @@ class ObjectApi(val objectService: ObjectService) {
         return ObjectsResponse(objectService.fetch().map { it.toResponse() })
     }
 
+    @GetMapping("recalculateValue")
+    fun recalculateValue(
+        @RequestParam("from", required = true) fromMeasure: String,
+        @RequestParam("to", required = true) toMeasure: String,
+        @RequestParam("value", required = true) value: String,
+        principal: Principal
+    ): ValueRecalculationResponse {
+        val username = principal.name
+        logger.debug("Recalculate value request by $username")
+        return ValueRecalculationResponse(
+            targetMeasure = toMeasure,
+            value = objectService.recalculateValue(fromMeasure, toMeasure, BigDecimal(value)).stripTrailingZeros().toPlainString()
+        )
```

Comment:
Provided we encapsulate BigDecimal into DecimalNumber all references to BigDecimal methods like stripTrailingZeros should be located inside DecimalNumber. 
The same applies to BigDecimal constructor.